src/nvim/option.c

```diff
@@ -2538,11 +2538,11 @@ did_set_string_option (
       if (s[2] == NUL)
         break;
     }
-  }
-  /* 'highlight' */
-  else if (varp == &p_hl) {
-    if (highlight_changed() == FAIL)
-      errmsg = e_invarg;        /* invalid flags */
+  } else if (varp == &p_hl) {
+    // 'highlight'
+    if (strcmp((char *)(*varp), HIGHLIGHT_INIT) != 0) {
+      errmsg = e_unsupportedoption;
```

Comment:
Is this needed here? Only in very special cases would someone needed to change this option, like someone forced to use a vim build without +syntax. Compare with `&encoding`, we do not say "change fileencoding to edit files in different encodings" in the error message.